ballerina-tests/tests/37_interfaces.bal

```diff
@@ -94,3 +94,47 @@ isolated function testInterfacesWithInterfaceTypeArray() returns error? {
     json expectedPayload = check getJsonContentFromFile("interfaces_with_interface_type_array.json");
     assertJsonValuesWithOrder(actualPayload, expectedPayload);
 }
+
+@test:Config {
+    groups: ["interfaces", "fragments"]
+}
+isolated function testInterfaceImplementingInterfacByQueryingOnInterface() returns error? {
+    string document = check getGraphQLDocumentFromFile("interfaces_implementng_interface_querying_on_interface.graphql");
+    string url = "http://localhost:9089/interfaces_implementing_interface";
+    json actualPayload = check getJsonPayloadFromService(url, document);
+    json expectedPayload = check getJsonContentFromFile("interfaces_implementng_interface_querying_on_interface.json");
+    assertJsonValuesWithOrder(actualPayload, expectedPayload);
+}
+
+@test:Config {
+    groups: ["interfaces", "fragments"]
+}
+isolated function testInterfaceImplementingInterfacByQueryingOnTransitiveType() returns error? {
+    string document = check getGraphQLDocumentFromFile("interfaces_implementng_interface_querying_on_transitive_type.graphql");
+    string url = "http://localhost:9089/interfaces_implementing_interface";
+    json actualPayload = check getJsonPayloadFromService(url, document);
+    json expectedPayload = check getJsonContentFromFile("interfaces_implementng_interface_querying_on_transitive_type.json");
+    assertJsonValuesWithOrder(actualPayload, expectedPayload);
+}
+
+@test:Config {
+    groups: ["interfaces", "fragments"]
+}
+isolated function testInterfaceImplementingInterfacByQueryingOnBothTransitiveTypeAndInterface() returns error? {
+    string document = check getGraphQLDocumentFromFile("interfaces_implementng_interface_querying_on_transitive_type_and_interface.graphql");
+    string url = "http://localhost:9089/interfaces_implementing_interface";
+    json actualPayload = check getJsonPayloadFromService(url, document);
+    json expectedPayload = check getJsonContentFromFile("interfaces_implementng_interface_querying_on_transitive_type_and_interface.json");
+    assertJsonValuesWithOrder(actualPayload, expectedPayload);
+}
+
+@test:Config {
+    groups: ["interfaces", "fragments"]
+}
+isolated function testInterfaceImplementingInterfacByQueryingFragmentOnInterface() returns error? {
+    string document = check getGraphQLDocumentFromFile("interfaces_implementng_interface_querying_fragement_on_transitive_interface.graphql");
+    string url = "http://localhost:9089/interfaces_implementing_interface";
+    json actualPayload = check getJsonPayloadFromService(url, document);
+    json expectedPayload = check getJsonContentFromFile("interfaces_implementng_interface_querying_fragement_on_transitive_interface.json");
+    assertJsonValuesWithOrder(actualPayload, expectedPayload);
+}
```

Comment:
Shall we add an introspection query for this service to check the information about the interfaces?

Reply:
Added